### DISCUSSION TITLE
isGesture flag for Apple Maps

### DISCUSSION THREAD

**Discussion by RyanTG** [react-native-maps/react-native-maps#4440] [Feature requests]
# Feature request

The `isGesture` flag currently only works for Google Maps. This request is for this property (or something like it) to work for Apple Maps, too.

## Why it is needed

To improve feature parity between and Apple Maps and Google Maps.

The "why did the camera move - user or automatic movement?" issue has been brought up at least half a dozen times over the last few years, with many users commenting that it is a much-needed feature. There are many events that cause the map to move on its own without direct pan/zoom inputs from the user (centering after clicking a marker, navigating back to the map, etc). But if you want to "refresh the results shown when the map is moved" then it's important that the refresh doesn't fire from every hair trigger automatic movement. The `isGesture` property solved this issue - but only for Google Maps!

In our particular case, we use `isGesture` to display a "search here" button, similar to what Google Maps currently shows on their map when you pan around. This buttons helps manage the rate at which the map refreshes as users pan around. If anyone happens to have an example of using a button like this on Apple Maps to detect user movements (without the `isGesture` flag) then I'd definitely like to see it!

Previous PRs that did not get merged:

https://github.com/react-native-maps/react-native-maps/pull/4322 (note that from the feedback, this one seems headed toward the right track, but it didn't capture all cases correctly).
https://github.com/react-native-maps/react-native-maps/pull/4113

Commit for the Google Maps implementation (merged): https://github.com/react-native-maps/react-native-maps/commit/84cd0c439c38d1757abdf9dae690b11488f214db

Previous Issue I created requesting this: https://github.com/react-native-maps/react-native-maps/issues/3952

**Comment by mmamani93** [comment]
I need to enable a button when user moves through the map. Currently using `onRegionChangeComplete` and there is no way if the map movement was a user interaction or a map animation. My app allows user to pick which map (Google or Apple) map they want to use so I have one less feature for Apple maps as we are today 😓

**Comment by lukeinage** [comment]
This is the main thing stopping my iOS application making use of apple maps

**Comment by statico** [comment]
I'm moving us to Google Maps because of this. As stated, if your app does a search to show items within the map area, the ability to differentiate between user-initiated map movement and automated movement is essential to prevent loops and other nasty behavior.

Some comparisons:
- The Airbnb app for iOS uses Google Maps, presumably for this reason
- The Yelp app for iOS appears to use Apple Maps but has a manual "Redo Search in This Area" button you need to tap
- Another iOS app I have uses MapBox

I'm going to consider switching to MapBox since it will be substantially cheaper than Google Maps.

**Reply by RyanTG** [reply]
> I'm going to consider switching to MapBox since it will be substantially cheaper than Google Maps.

The Google Maps Native Mobile Dynamic SDK is free https://developers.google.com/maps/billing-and-pricing/pricing#mobile-dynamic

You do not need to be concerned with costs unless you are using some kind of custom cloud-based styles for it (that is the only instance where costs are incurred). 

I'm the OP in this discussion, and we're actually almost done moving off of react-native-maps to rnmapbox. This isGesture issue was not part of the decision. Rather, these issues were:

https://github.com/react-native-maps/react-native-maps/issues/4625

All the issues about using custom markers in Android being very slow.

Try as we might, we could not overcome those two issues, so we went nuclear. rnmapbox has differences and challenges. But it does not have the same issues with identifying user gestures, performance, and rendering many markers.

**Reply by statico** [reply]
> The Google Maps Native Mobile Dynamic SDK is free https://developers.google.com/maps/billing-and-pricing/pricing#mobile-dynamic

🤦 Of course. Thank you for the correction. In that case we can stick with react-native-maps.
